operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ServiceOperator.java

```diff
@@ -70,6 +78,31 @@ public ServiceOperator(Vertx vertx, KubernetesClient client) {
         }
     }
 
+    /**
+     * Finds annotations managed by the Rancher cattle agents (if any), and merge them into the desired spec.
+     *
+     * This makes sure there is no infinite loop where Rancher tries to add annotations, while Rancher keeps
+     * removing them during reconciliation.
+     *
+     * @param current   Current Service
+     * @param desired   Desired Service
+     */
+    private void patchAnnotations(Service current, Service desired) {
+        Map<String, String> currentAnnotations = current.getMetadata().getAnnotations();
+        if (currentAnnotations != null) {
+            Map<String, String> matchedAnnotations = currentAnnotations.keySet().stream()
+                    .filter(annotation -> LOADBALANCER_ANNOTATION_WHITELIST.stream().anyMatch(whitelist -> whitelist.test(annotation)))
```

Comment:
I wonder if we really need the matching here. If the `contains` or `startsWith` which was used originally wasn't better in terms of performance etc. @tombentley WDYT?

Reply:
I replaced the regex'es with `startsWith` :-)